app/models/rr_pending_change.rb

```diff
@@ -14,10 +14,7 @@ def self.last_id
   class << self; alias_method :backlog, :count; end
 
   def self.backlog_details
-    counts = all(
-      :select => "change_table, COUNT(id) AS count_all",
-      :group  => "change_table"
-    )
-    counts.each_with_object({}) { |c, h| h[c.change_table] = c.count_all.to_i }
+    counts = select("change_table, COUNT(id) AS count_all").group("change_table").order('count_all DESC').limit(20).to_a
+    counts.each_with_object([]) { |c, a| a << [c.change_table, c.count_all.to_i] }
```

Comment:
Oh I didn't notice that the original code was each_with_object with a *Hash*.  I like the original code for this line for the reasons @matthewd gave.

@matthewd I was more saying that each_with_object with an *Array* is an anti-pattern, as that almost always drops down to collect/select/flat_map

And yes, Hashes are ordered.